src/unoserver/server.py

```diff
@@ -80,9 +76,27 @@ def main():
         default="libreoffice",
         help="The path to the LibreOffice executable",
     )
+    parser.add_argument(
+        "--user-installation",
+        default=None,
+        help="The path to the LibreOffice user profile"
+    )
     args = parser.parse_args()
 
-    server = UnoServer(args.interface, args.port)
+    server = UnoServer(
+        args.interface,
+        args.port,
+    )
+
+    if args.user_installation is None:
+        with tempfile.TemporaryDirectory() as tmpuserdir:
```

Comment:
This still means the directory is deleted before the server is even started. The call to the server must be within the context manager.

Reply:
And in fact the whole process and waiting needs to be inside the context manager in this case.

Reply:
Am I in the right place? [3131ef5](https://github.com/unoconv/unoserver/pull/76/commits/3131ef5058da38ca6168a6ad55ec20553a75a4c0)

Reply:
Yup, that looks good. I will take a closer look soon.